extra_data/components.py

```diff
@@ -639,3 +639,41 @@ class LPD1M(MPxDetectorBase):
     """
     _source_re = re.compile(r'(?P<detname>(.+)_LPD1M(.*))/DET/(\d+)CH')
     module_shape = (256, 256)
+
+
+def identify_mpx_detectors(data, detector_name=None, kind=None, single=False):
```

Comment:
I didn't polish this bit yet - also with the bare assert statement - because I thought we might still want to discuss exactly how it should work. Strings or classes? Do we want any fuzziness, e.g. allowing `LPD` instead of `LPD1M`?

Reply:
Probably string is a bit more flexible. I am thinking also to allow list of kinds (or comma separated items in a string)

Reply:
Yep, I can see a list being useful, where an operation makes sense for a subset of the supported detectors.

How much fuzziness should we allow? Should `AGIPD` mean `AGIPD1M`, even though there will one day be AGIPD4M? Or should there be some kind of hierarchy, so `AGIPD` covers both sizes?

Reply:
Part of me wants to keep the fuzziness minimal here and leave it up to the applications to add extra shortcuts. But that's just putting off the question, and it would be good to have a consistent scheme between different tools.

Reply:
Yes, I was thinking that AGIPD would cover both 1 and 4M. In practice there won't be any run with both detector in, but there might be the same code running on runs containing either of them.

Reply:
on the other side, it you think that this function is mostly for us to use, then I think I'd rather use classes (or list of classes) instead of strings.

Reply:
That's interesting. I initially wrote it to work with a class, but I figured that if you're exposing it in UI, you'll get a string from the user, and something has to convert that to a class, so why not do it in one place. But maybe we should leave the str -> class conversion up to tools, and just take a list of classes.

Reply:
For now I've restricted this to a list of classes to keep it simple - we can always add support for strings later.

Reply:
> Yes, I was thinking that AGIPD would cover both 1 and 4M. In practice there won't be any run with both detector in, but there might be the same code running on runs containing either of them.

Sorry for digging out older statements from a previous review stage. I am just interested in the scientific background, so it's a bit off-topic, but may have impact on the treatment in software:  
looking a schematic SPB instrument figures (instrumentation set-up), it always seemed to me that the parallel use of both AGIPD-1M and -4M in the same run is exactly the purpose. So that the far-away AGIPD-4M will resolve the low-angle diffraction better.